app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,342 @@
+package fr.free.nrw.commons.nearby
+
+import com.nhaarman.mockitokotlin2.*
+import fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao
+import fr.free.nrw.commons.location.LatLng
+import fr.free.nrw.commons.location.LocationServiceManager
+import fr.free.nrw.commons.location.LocationServiceManager.LocationChangeType
+import fr.free.nrw.commons.nearby.contract.NearbyParentFragmentContract
+import fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+
+/**
+ * The unit test class for NearbyParentFragmentPresenter
+ */
+class NearbyParentFragmentPresenterTest {
+    @Mock
+    internal lateinit var nearbyParentFragmentView: NearbyParentFragmentContract.View
+    @Mock
+    internal lateinit var bookmarkLocationsDao: BookmarkLocationsDao
+    @Mock
+    internal lateinit var latestLocation: LatLng
+    @Mock
+    internal lateinit var cameraTarget: LatLng
+    @Mock
+    internal lateinit var selectedLabels: List<Label>
+
+    private lateinit var nearbyPresenter: NearbyParentFragmentPresenter
+    private lateinit var mapboxCameraTarget: com.mapbox.mapboxsdk.geometry.LatLng
+
+    /**
+     * initial setup
+     */
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        nearbyPresenter = NearbyParentFragmentPresenter(bookmarkLocationsDao)
+        nearbyPresenter.attachView(nearbyParentFragmentView)
+    }
+
+    /**
+     * Tests nearby operations are initialized
+     */
+    @Test
+    fun testInitializeNearbyMapOperations() {
+        nearbyPresenter.initializeMapOperations()
+        verify(nearbyParentFragmentView).enableFABRecenter()
+        testPlacesPopulatedForLatestLocationWhenLocationSignificantlyChanged()
+        verify(nearbyParentFragmentView).addSearchThisAreaButtonAction()
```

Comment:
I am not sure about this line. @macgills can you guide me? When nearbyFragmentPresenter was a spy, this line was 
```
       verify(nearbyPresenter).updateMapAndList(LocationChangeType.LOCATION_SIGNIFICANTLY_CHANGED)
```
Now I changed it to this way, but doesn't seem like a best practice.

Reply:
You shouldn't generally be calling another test method but yo can make this DRY
```
whenever(nearbyParentFragmentView.isNetworkConnectionEstablished()).thenReturn(true)
        whenever(nearbyParentFragmentView.getLastLocation()).thenReturn(latestLocation)
        whenever(nearbyParentFragmentView.isCurrentLocationMarkerVisible()).thenReturn(false)
```
extract this to a function `expectMapAndListUpdate`

and then this should be the only place you are verifying
```
        if (locationChangeType.equals(LOCATION_SIGNIFICANTLY_CHANGED)
                || locationChangeType.equals(MAP_UPDATED)) {
            Timber.d("LOCATION_SIGNIFICANTLY_CHANGED");
            lockUnlockNearby(true);
            nearbyParentFragmentView.setProgressBarVisibility(true);
            nearbyParentFragmentView.populatePlaces(lastLocation);
``

for `LOCATION_SIGNIFICANTLY_CHANGED` but seeing as you have to verify this functionality for `MAP_UPDATED` as well you would also want to extract a method for this verification

Reply:
I reduced code repetitions, but still calling the test method directly instead of repeating it seemed to me like a cleaner way. So I still keep this line.
```testPlacesPopulatedForLatestLocationWhenLocationSignificantlyChanged ```

Your suggestion is changing this line with the content of the method:
```
        expectMapAndListUpdate()
        nearbyPresenter.updateMapAndList(LocationChangeType.LOCATION_SIGNIFICANTLY_CHANGED)
        verify(nearbyParentFragmentView).disableFABRecenter();
        verify(nearbyParentFragmentView).setProgressBarVisibility(true)
        verify(nearbyParentFragmentView).populatePlaces(latestLocation)
```
if I am not mistaken, right?

Reply:
Well, now I used the method content instead. Please ignore my previous message.